### PR TITLE
[RFC] move chroot/bootstrap packages to bootstrap repository

### LINKED ISSUE #39166: [RFC] xbps repository for internal xbps-src and bootstrap packages
It might be nice to create a xbps repository for packages that are supposed to be used by xbps-src only, like `-bootstrap` packages and maybe `chroot-*` packages.

This would avoid them showing up in default repository and avoids users installing `chroot-*` packages due to virtual packages by accident.

We would add `repository="bootstrap"` to templates and add that repository to the repository pool in `xbps-src`.

### PR DESCRIPTION
**Pull request by classabbyamp**
closes #39166

#### Testing the changes
- I tested the changes in this PR: **NO**
    - have not tested this yet, will do so soon

[ci skip]

### CODE REVIEW

**Comment by ahesford** [comment]
I like the idea of a separate bootstrap repo.

Even better would be a `void-packges` hierarchy that defines destination repositories instead of a `repository` template argument. For example: move all `repository=nonfree` templates under `nonfree` rather than `srcpkgs`, move the new `repository=bootstrap` templates under `bootstrap`, and we could even do away with `restricted` by creating a `restricted` hierarchy.

All of this is out of scope for this proposal, but I can dream.

File: common/xbps-src/shutils/chroot.sh [hunk 1]
```diff
@@ -210,12 +210,14 @@ chroot_sync_repodata() {
     mkdir -p $confdir
     if [ -n "$XBPS_ALT_REPOSITORY" ]; then
         ( \
+            echo "repository=$hostdir/binpkgs/${XBPS_ALT_REPOSITORY}/bootstrap"; \
```
Comment (sgn) [inline common/xbps-src/shutils/chroot.sh:213]: While we're at it, can we change this to heredoc? subshell is not necessary, I think

**Comment by classabbyamp** [comment]
here's a list of all files from bootstrap/chroot packages on the repos: [bootstrap-files.txt](https://github.com/void-linux/void-packages/files/10048259/bootstrap-files.txt)

[generated with](https://github.com/classabbyamp/xX_tools_Xx/blob/tools/xrls) `xrls chroot-bash chroot-distcc chroot-gawk chroot-git chroot-grep chroot-util-linux chroot-curl cargo-bootstrap cargo-auditable-bootstrap dmd-bootstrap go1.12-bootstrap musl-bootstrap rust-bootstrap openjdk7-bootstrap openjdk9-bootstrap openjdk10-bootstrap openjdk12-bootstrap openjdk13-bootstrap openjdk14-bootstrap openjdk15-bootstrap openjdk16-bootstrap`

**Comment by classabbyamp** [comment]
files to be removed from repos (old, removed bootstrap packages): [rm-files.txt](https://github.com/void-linux/void-packages/files/10502020/rm-files.txt)

files to be moved to their respective bootstrap repos (`/current{,/bootstrap}`, `/current/aarch64{,/bootstrap}`, `/current/multilib{,/bootstrap}`, `/current/musl{,/bootstrap}`): [bootstrap-files.txt](https://github.com/void-linux/void-packages/files/10502132/bootstrap-files.txt)

**Comment by the-maldridge** [comment]
github is being dumb, this was pushed by hand as a2c510b8a4

**Pull request closed without merging**